nuxeo-services/nuxeo-permissions/src/main/java/org/nuxeo/ecm/permissions/PermissionGrantedNotificationListener.java

```diff
@@ -156,8 +156,16 @@ protected void handleEvent(Event event) {
             }
 
             OperationChain chain = new OperationChain("SendMail");
-            chain.add(SendMail.ID).set("from", from).set("to", to).set("HTML", true).set("subject", subject)
-                    .set("message", ACE_GRANTED_TEMPLATE);
+
+            // if there is more than one recipient use "bcc:" instead of "to:" for avoiding email sharing
+            final boolean useBcc = to.size() > 1;
```

Comment:
No need for `final`.

Reply:
I know, it's just that I come from the Scala world and just try to enforce immutability everywhere, but does not make any difference here. It's a old habit that i'll try to avoid.